### PR TITLE
fix: setItems type fixed in CartProviderState

### PR DESCRIPTION
**Pull request by Skidragon**

### CODE REVIEW

**Comment by Skidragon** [comment]
The reason why I made a change to the setItems function is because I was getting an NaN for totalItems.

**Comment by notrab** [comment]
Thanks for the PR @Skidragon!

Are you happy for me mark this ready for review?

**Comment by Skidragon** [comment]
@notrab yep!

**Comment by Skidragon** [comment]
@notrab Should we implicitly add a quantity of 1 to each item in setItems? I noticed in defaultItems doesn't have a default quantity added to the item.

**Comment by notrab** [comment]
Sounds good! I don’t think that would cause any backwards compatibility issues so feel free to add it to this PR and I’ll get it merged 🔥

**Comment by Skidragon** [comment]
@notrab I made the update to the documentation, I think the CartProvider defaultItems may need to have a quantity of 1 implicitly set also for constancy and the docs doesn't talk about it either but I believe it should be for a separate PR since this is for setItems only and don't want to mix concerns?

**Comment by notrab** [comment]
@Skidragon sure.

Is this PR ready for review now? 😄

File: src/index.tsx [hunk 1]
```diff
@@ -191,7 +192,12 @@ export const CartProvider: React.FC<{
   const setItems = (items: Item[]) => {
     dispatch({
       type: "SET_ITEMS",
-      payload: items,
+      payload: items.map(item => {
```
Comment (notrab) [inline src/index.tsx:195]:
```suggestion
      payload: items.map(item => ({
```

File: src/index.tsx [hunk 2]
```diff
@@ -191,7 +192,12 @@ export const CartProvider: React.FC<{
...
+        return {
```
Comment (notrab) [inline src/index.tsx:196]:
```suggestion
```

File: src/index.tsx [hunk 3]
```diff
@@ -191,7 +192,12 @@ export const CartProvider: React.FC<{
...
+          ...item,
+          quantity: item.quantity || 1
+        }
```
Comment (notrab) [inline src/index.tsx:199]:
```suggestion
```

File: src/index.tsx [hunk 4]
```diff
@@ -191,7 +192,12 @@ export const CartProvider: React.FC<{
...
+      }),
```
Comment (notrab) [inline src/index.tsx:200]:
```suggestion
      })),
```

**Comment by Skidragon** [comment]
> @Skidragon sure.
> 
> Is this PR ready for review now? 😄 

Lol yeah. I got nervous and just made sure the tests covered everything.

**Review by notrab — COMMENTED** [review]
Few a small thing, I think I used this approach when only returning an object elsewhere in the lib.

**Comment by Skidragon** [comment]
@notrab the small things have been added :)

**Pull request merged**